netplugin/cluster/cluster.go

```diff
@@ -100,8 +106,45 @@ func httpPost(url string, req interface{}, resp interface{}) error {
 	return nil
 }
 
+// getMasterLockHolder returns the IP of current master lock hoder
+func getMasterLockHolder() (string, error) {
+	// Create an objdb client
+	objdbClient := objdb.NewClient("")
+
+	// Create the lock
+	leaderLock, err := objdbClient.NewLock("netmaster/leader", "", 0)
+	if err != nil {
+		log.Fatalf("Could not create leader lock. Err: %v", err)
+	}
+
+	// get current holder of leader lock
+	masterNode := leaderLock.GetHolder()
+	if masterNode == "" {
+		log.Errorf("No leader node found")
+		return "", errors.New("No leader node")
+	}
+
+	return masterNode, nil
+}
+
 // MasterPostReq makes a POST request to master node
 func MasterPostReq(path string, req interface{}, resp interface{}) error {
+	// first find the holder of master lock
+	masterNode, err := getMasterLockHolder()
```

Comment:
Don't we need a similar change to the rpc call from ofnet agent as well? Did I miss it in the diff?